### ISSUE TITLE
Fix bug in "nice" bin boundary calculation.

### ISSUE THREAD

**Issue by domoritz** [vega/vega#925]
![image](https://user-images.githubusercontent.com/589034/28232890-fd22df52-68a7-11e7-9aa3-ee5a113b9fde.png)

Previous Vega-Lite issue: https://github.com/vega/vega-lite/issues/2660

```json
{
  "$schema": "http://vega.github.io/schema/vega/v3.0.json",

  "data": [
    {
      "name": "source_0",
      "values": [
        {
          "Carrier": "AA",
          "Average of Distance": 569,
          "Average of DepTime": 1084.784,
          "_id": 155250
        },
        {
          "Carrier": "AA",
          "Average of Distance": 646,
          "Average of DepTime": 1291.4252873563219,
          "_id": 155251
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1345,
          "Average of DepTime": 892.6491228070175,
          "_id": 155823
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1671,
          "Average of DepTime": 818,
          "_id": 155824
        },
        {
          "Carrier": "AA",
          "Average of Distance": 601,
          "Average of DepTime": 909.3888888888889,
          "_id": 155825
        },
        {
          "Carrier": "AA",
          "Average of Distance": 573,
          "Average of DepTime": 1253.877192982456,
          "_id": 155826
        },
        {
          "Carrier": "AA",
          "Average of Distance": 600,
          "Average of DepTime": 1261.6241134751774,
          "_id": 155827
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1017,
          "Average of DepTime": 1391.588888888889,
          "_id": 155828
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1120,
          "Average of DepTime": 1387.0873015873017,
          "_id": 155829
        },
        {
          "Carrier": "AA",
          "Average of Distance": 993,
          "Average of DepTime": 1203.715909090909,
          "_id": 155830
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2077,
          "Average of DepTime": 1559.8307692307692,
          "_id": 155831
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1171,
          "Average of DepTime": 1167.048872180451,
          "_id": 155832
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2446,
          "Average of DepTime": 851.25,
          "_id": 155833
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2267,
          "Average of DepTime": 1969.7179487179487,
          "_id": 155834
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1723,
          "Average of DepTime": 1134.6601941747572,
          "_id": 155835
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2370,
          "Average of DepTime": 882.390243902439,
          "_id": 155836
        },
        {
          "Carrier": "AA",
          "Average of Distance": 304,
          "Average of DepTime": 1291.046052631579,
          "_id": 155837
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1095,
          "Average of DepTime": 1318.7125,
          "_id": 155838
        },
        {
          "Carrier": "AA",
          "Average of Distance": 247,
          "Average of DepTime": 1265.9335443037974,
          "_id": 155839
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1143,
          "Average of DepTime": 617.12,
          "_id": 155840
        },
        {
          "Carrier": "AA",
          "Average of Distance": 733,
          "Average of DepTime": 1605.5555555555557,
          "_id": 155841
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2279,
          "Average of DepTime": 1563.5079365079366,
          "_id": 155842
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1660,
          "Average of DepTime": 962.9090909090909,
          "_id": 155843
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2422,
          "Average of DepTime": 1073.16,
          "_id": 155844
        },
        {
          "Carrier": "AA",
          "Average of Distance": 954,
          "Average of DepTime": 2036.1304347826087,
          "_id": 155845
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2724,
          "Average of DepTime": 2307.451612903226,
          "_id": 155846
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1721,
          "Average of DepTime": 877.5376344086021,
          "_id": 155847
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2378,
          "Average of DepTime": 872.3030303030303,
          "_id": 155848
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1107,
          "Average of DepTime": 1265.9416666666666,
          "_id": 155849
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2296,
          "Average of DepTime": 1141.7567567567567,
          "_id": 155850
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1464,
          "Average of DepTime": 1011.3771626297578,
          "_id": 155851
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2586,
          "Average of DepTime": 1344.1184210526317,
          "_id": 155852
        },
        {
          "Carrier": "AA",
          "Average of Distance": 337,
          "Average of DepTime": 1384.5714285714287,
          "_id": 155853
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2585,
          "Average of DepTime": 1537.2727272727273,
          "_id": 155854
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1846,
          "Average of DepTime": 1191.9932432432433,
          "_id": 155855
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2521,
          "Average of DepTime": 1366.6767676767677,
          "_id": 155856
        },
        {
          "Carrier": "AA",
          "Average of Distance": 651,
          "Average of DepTime": 1131.3538461538462,
          "_id": 155857
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1438,
          "Average of DepTime": 952.2719298245614,
          "_id": 155858
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1829,
          "Average of DepTime": 1289.6486486486488,
          "_id": 155859
        },
        {
          "Carrier": "AA",
          "Average of Distance": 621,
          "Average of DepTime": 1261.1020408163265,
          "_id": 155860
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1475,
          "Average of DepTime": 1505.4175824175825,
          "_id": 155861
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2165,
          "Average of DepTime": 1198.1935483870968,
          "_id": 155862
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1598,
          "Average of DepTime": 1158.1896551724137,
          "_id": 155863
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1045,
          "Average of DepTime": 1305.0884955752213,
          "_id": 155864
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2072,
          "Average of DepTime": 1566.741935483871,
          "_id": 155865
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1576,
          "Average of DepTime": 1340.7857142857142,
          "_id": 155866
        },
        {
          "Carrier": "AA",
          "Average of Distance": 1727,
          "Average of DepTime": 99.2258064516129,
          "_id": 155867
        },
        {
          "Carrier": "AA",
          "Average of Distance": 989,
          "Average of DepTime": 1046.3483146067415,
          "_id": 155868
        },
        {
          "Carrier": "AA",
          "Average of Distance": 2088,
          "Average of DepTime": 158.66666666666666,
          "_id": 155869
        },
        {
          "Carrier": "AS",
          "Average of Distance": 1616,
          "Average of DepTime": 1651.5806451612902,
          "_id": 155952
        },
        {
          "Carrier": "AS",
          "Average of Distance": 1660,
          "Average of DepTime": 1372.7608695652175,
          "_id": 155953
        },
        {
          "Carrier": "AS",
          "Average of Distance": 1927,
          "Average of DepTime": 1659.1333333333334,
          "_id": 155954
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2434,
          "Average of DepTime": 1587.5666666666666,
          "_id": 155955
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2425,
          "Average of DepTime": 1906.4193548387098,
          "_id": 155956
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2402,
          "Average of DepTime": 1335.24,
          "_id": 155957
        },
        {
          "Carrier": "AS",
          "Average of Distance": 261,
          "Average of DepTime": 998.8387096774194,
          "_id": 155958
        },
        {
          "Carrier": "AS",
          "Average of Distance": 1533,
          "Average of DepTime": 1092.7477477477478,
          "_id": 155959
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2717,
          "Average of DepTime": 1580.1860465116279,
          "_id": 155960
        },
        {
          "Carrier": "AS",
          "Average of Distance": 224,
          "Average of DepTime": 931.2689075630252,
          "_id": 155961
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2777,
          "Average of DepTime": 2075.548387096774,
          "_id": 155962
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2409,
          "Average of DepTime": 1498.5384615384614,
          "_id": 155963
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2603,
          "Average of DepTime": 1230.0333333333333,
          "_id": 155964
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2614,
          "Average of DepTime": 1668.8076923076924,
          "_id": 155965
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2677,
          "Average of DepTime": 1123.967741935484,
          "_id": 155966
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2417,
          "Average of DepTime": 1521.8333333333333,
          "_id": 155967
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2306,
          "Average of DepTime": 1835.032258064516,
          "_id": 155968
        },
        {
          "Carrier": "AS",
          "Average of Distance": 1874,
          "Average of DepTime": 1297.1132075471698,
          "_id": 155969
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2422,
          "Average of DepTime": 716.0967741935484,
          "_id": 155970
        },
        {
          "Carrier": "AS",
          "Average of Distance": 571,
          "Average of DepTime": 1354.5625,
          "_id": 155971
        },
        {
          "Carrier": "AS",
          "Average of Distance": 234,
          "Average of DepTime": 717.96,
          "_id": 155972
        },
        {
          "Carrier": "AS",
          "Average of Distance": 123,
          "Average of DepTime": 1347.9032258064517,
          "_id": 155973
        },
        {
          "Carrier": "AS",
          "Average of Distance": 909,
          "Average of DepTime": 1303.0224719101125,
          "_id": 155974
        },
        {
          "Carrier": "AS",
          "Average of Distance": 95,
          "Average of DepTime": 1518.8833333333334,
          "_id": 155975
        },
        {
          "Carrier": "AS",
          "Average of Distance": 198,
          "Average of DepTime": 994.4137931034483,
          "_id": 155976
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2874,
          "Average of DepTime": 1030.3333333333333,
          "_id": 155977
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2732,
          "Average of DepTime": 1263.25,
          "_id": 155978
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2378,
          "Average of DepTime": 1616.0967741935483,
          "_id": 155979
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2607,
          "Average of DepTime": 1156.9333333333334,
          "_id": 155980
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2556,
          "Average of DepTime": 1456.7272727272727,
          "_id": 155981
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2688,
          "Average of DepTime": 1494.8064516129032,
          "_id": 155982
        },
        {
          "Carrier": "AS",
          "Average of Distance": 2384,
          "Average of DepTime": 1497,
          "_id": 155983
        },
        {
          "Carrier": "AS",
          "Average of Distance": 234,
          "Average of DepTime": 2054.52,
          "_id": 155984
        },
        {
          "Carrier": "AS",
          "Average of Distance": 873,
          "Average of DepTime": 1538.2285714285715,
          "_id": 156043
        },
        {
          "Carrier": "AS",
          "Average of Distance": 933,
          "Average of DepTime": 1206.5934065934066,
          "_id": 156044
        },
        {
          "Carrier": "AS",
          "Average of Distance": 129,
          "Average of DepTime": 677.1666666666666,
          "_id": 156045
        }
      ]
    },
    {
      "name": "data_0",
      "source": "source_0",
      "transform": [
        {
          "type": "extent",
          "field": "Average of DepTime",
          "signal": "bin_maxbins_64_Average_of_DepTime_extent"
        },
        {
          "type": "bin",
          "field": "Average of DepTime",
          "as": [
            "bin_maxbins_64_Average of DepTime_start",
            "bin_maxbins_64_Average of DepTime_end"
          ],
          "signal": "bin_maxbins_64_Average_of_DepTime_bins",
          "maxbins": 64,
          "extent": {
            "signal": "bin_maxbins_64_Average_of_DepTime_extent"
          }
        }
      ]
    }
  ],
  "signals": [
    {
      "name": "height",
      "update": "500"
    }
  ],
  "scales": [
    {
      "name": "y",
      "type": "linear",
      "domain": {
        "data": "data_0",
        "field": "bin_maxbins_64_Average of DepTime_start"
      },
      "range": "height"
    }
  ],
  "axes": [
    {
      "title": "Average of DepTime",
      "scale": "y",
      "orient": "left",
      "values": {
        "signal": "sequence(bin_maxbins_64_Average_of_DepTime_bins.start, bin_maxbins_64_Average_of_DepTime_bins.stop + bin_maxbins_64_Average_of_DepTime_bins.step, bin_maxbins_64_Average_of_DepTime_bins.step)"
      },
      "zindex": 1
    }
  ]
}
```

**Comment by jheer** [comment]
Thanks! The offending line appears to be [bin.js#44](https://github.com/vega/vega-statistics/blob/master/src/bin.js#L44). I think the nice calculation is producing a candidate _start_ value that is greater than the minimum extent value, hence the (un-niced) min value is being returned as the bin _start_, while the bin _end_ and _step_ have the calculated nice values. If so, one solution might be to decrement the "nice" _start_ by one _step_ rather than use the current incorrect min filter.

**Comment by jheer** [comment]
Fixed in vega-statistics v1.1.4.
